### PR TITLE
Add feature for Streams API

### PR DESCRIPTION
**Pull request by ksm2**
Hi, I added a base feature JSON for the WHATWG Streams API, related to issue #2042. I am not quite firm with adding tests for caniuse, maybe someone can help me with that?

### CODE REVIEW

File: features-json/streams.json [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+      "11.5":"u",
+      "12":"u",
+      "12.1":"u",
+      "37":"u"
+    },
+    "and_chr":{
+      "61":"u"
+    },
+    "and_ff":{
+      "55":"u"
```
Comment (lpd-au) [inline features-json/streams.json]: Could match FF55 desktop as `n` right?

File: features-json/streams.json [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+    },
+    "ie_mob":{
+      "10":"u",
```
Comment (lpd-au) [inline features-json/streams.json]: Ditto for IE10 desktop

File: features-json/streams.json [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+      "1.2":"u"
+    },
+    "baidu":{
+      "7.12":"u"
+    }
+  },
+  "notes":"",
+  "notes_by_num":{
+    "1": "IE implements a different API than the one from WHATWG.",
+    "2": "Chrome is still in development."
```
[the remark this answers is not in the document]
Comment (ksm2) [reply features-json/streams.json]: OK, I just thought that one should maybe differ in the Streams API support levels. If you take a look [at the MDN browser compatibility](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream#Browser_Compatibility), they differentiate between basic support and `pipeThrough` / `pipeTo` support. I would also like to add a test case for BYOB streams which are still not supported in current Chrome 61. Wouldn't it be good to have two notes therefore?

[the remark this answers is not in the document]
Comment (lpd-au) [reply features-json/streams.json]: Assuming the pipes weren't behind a separate flag then, 52-58 could be `a #2` with the note specifically explaining the lack of pipe support. Then, if it's appropriate, have a different note for 59+ explaining the lack of BYOB streams.

**Comment by ksm2** [comment]
@lpd-au I wrote some tests and put them into the following gist: https://gist.github.com/ksm2/890bf989629e1afe6b704fdfae3f97aa

"streams0" returns true for basic support (afaik stable Chrome & Opera and Firefox with flags `javascript.options.streams` and `dom.streams.enabled` set to true)
"streams1" returns true for `WritableStream` support (Chrome & Opera, no Firefox)
"streams2" returns true for `pipeThrough` and `pipeThrough` (as "streams0")
"streams3" returns true for BYOB support (Firefox with flags `javascript.options.streams` and `dom.streams.enabled` set to true)

**Comment by Fyrd** [comment]
Thanks!

**Pull request merged**